app/models/category_usage.rb

```diff
@@ -0,0 +1,45 @@
+class CategoryUsage < ApplicationRecord
+  belongs_to :category
+
+
+  def self.store_chosen_categories(s_query, chosen_categories, s_key = "")
+    match_search_query_to_categories s_query, s_key unless s_query.blank?
+    if !chosen_categories.blank?
+      cat_ids = get_chosen_category_ids(chosen_categories)
+      cat_ids.each do |cat_id|
+        CategoryUsage.create(category_id: cat_id, key: s_key)
+      end
+    end
+  end
+
+  def self.get_chosen_category_ids(chosen_categories)
+    cat_ids = []
+    chosen_categories.each do |chosen_cat|
+      cat_rec = Category.where(:name => chosen_cat).first
```

Comment:
we can do a `find_by` instead and push that to the `cat_ids` array and in line 21 do a `cat_ids.compact` and I think that removes the `nil` values

Reply:
Yep  - that works better.